---
title: "Week11"
author: "Uwaysah"
date: "2023-05-15"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Section 1. Proportion of G/G in a population

Downloaded a CSV file from Ensemble < 
https://useast.ensembl.org/Homo_sapiens/Variation/Sample?db=core;r=17:39894595-39895595;v=rs8067378;vdb=variation;vf=105535077#373531_tablePanel

Here we read this CSV file

```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```
```{r}
table(mxl$Genotype..forward.strand.)
```
```{r}
table(mxl$Genotype..forward.strand.) / nrow(mxl) * 100
```
```{r}
gbr <- read.csv("373522-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(gbr)
```
```{r}
round(table(gbr$Genotype..forward.strand.) / nrow(gbr) *100 , 2)
```
This variant that is associated with childhood asthma is more frequent in the GBR population than the MXL population. 

#section 4: Population Scale Analysis
One sample is obviously not enough to know what is happening in a population. You are interested in assessing genetic differences on a population scale. 

How many samples do we have?
```{r}
expr <- read.table("rs8067378_ENSG00000172057.6.txt")
head(expr)
```
```{r}
nrow(expr)
```
```{r}
table(expr$geno)
```
```{r}
library(ggplot2)
```

Let's make a boxplot

```{r}
ggplot(expr) + aes(geno,exp, fill=geno) + geom_boxplot(notch=TRUE)
```

>Q1: What are those 4 candidate SNPs?

>Q2: What three genes do these variants overlap or effect? 

>Q3: What is the location of rs8067378 and what are the different alleles for rs8067378?

>Q4: Name at least 3 downstream genes for rs8067378?  

>Q5: What proportion of the Mexican Ancestry in Los Angeles sample population (MXL) are homozygous for the asthma associated SNP (G|G)?  

>Q6. Back on the ENSEMBLE page, use the “search for a sample” field above to find the particular sample HG00109. This is a male from the GBR population group. What is the genotype for this sample?

>Q7: How many sequences are there in the first file? What is the file size and format of the data? Make sure the format is fastqsanger here!

>Q8: What is the GC content and sequence length of the second fastq file?

>Q9: How about per base sequence quality? Does any base have a mean quality score below 20?

>Q10: Where are most the accepted hits located?

>Q11: Following Q10, is there any interesting gene around that area?

>Q12: Cufflinks again produces multiple output files that you can inspect from your right-handside galaxy history. From the “gene expression” output, what is the FPKM for the ORMDL3 gene? What are the other genes with above zero FPKM values? 
